src/config.cpp

```diff
@@ -2278,10 +2278,14 @@ static int updateMaxclients(long long val, long long prev, const char **err) {
             if ((unsigned int) aeGetSetSize(g_pserver->rgthreadvar[iel].el) <
                 g_pserver->maxclients + CONFIG_FDSET_INCR)
             {
-                if (aeResizeSetSize(g_pserver->rgthreadvar[iel].el,
-                    g_pserver->maxclients + CONFIG_FDSET_INCR) == AE_ERR)
-                {
-                    *err = "The event loop API used by Redis is not able to handle the specified number of clients";
+                int res = aePostFunction(g_pserver->rgthreadvar[iel].el, [iel] {
+                    aeResizeSetSize(g_pserver->rgthreadvar[iel].el, g_pserver->maxclients + CONFIG_FDSET_INCR);
```

Comment:
At a minimum we need to do something useful with the aeResizeSetSize() return code for teh current thread.  Checking the return of aePostFunction is not sufficient to know the update will actually work.

Reply:
Thanks for your feedback John.

aePostFunction (with default values) will always return "**AE_OK**" for the current thread  after it successfully call the "aeResizeSetSize" function. It seems there is no way to get the return code for aeResizeSetSize call. 

I tried to call aePostFunction with "synchronous = true" , but, client(cli) waits forever for the reply.  Is there  a way to get immediate execution on each event-loop? 

However, Each threads can check the return code of "aeResizeSetSize" call. Based on that, I implemented the error handling. Please let me know your thoughts on this changes.

Reply:
@JohnSully aeResizeSetSize() return code for the current thread is properly handled while serverLog is updated for all other threads.

Please let me know your thoughts on this changes